SECURITY.md

```diff
@@ -29,3 +35,89 @@ If you are editing this document please feel encouraged to change this format to
 - #993 Add restricted permissions to RocksDB - 1.4.5 - Anonymous security review
 - #740 Add hardening compile flags and `-fPIE` - 1.4.1 - Anonymous security review
 - #300 Add restricted permissions to osqueryd logging - 1.0.4
+
+## Security Requirements Met by Functionality
```

Comment:
This header doesn't feel meaningful to me. It feels like some very fancy language someone wrote a phd for. (OTOH, maybe it's industry standard)

I think we should try to find a plain language way to express what we mean. 

And I'm not sure if we're trying to express:
1. These are the basic principles of osquery development, which contribute to our security
2. These are specific threat case we have thought about, written up, and mitigated

Reply:
I get what you're saying, and I'm renaming this heading `Security Design Considerations of osquery`

Reply:
In that context, do "goals" or "tenants" make sense?

Reply:
To me, `tenants` doesn't carry any meaning in this context. I like `goals`, but `considerations` gives us the flexibility to document the explicitly unaddressed threats in our model, like, we know a root attacker could kill the osquery process and we have chosen to place that out of scope. It has been _considered_ and here's our thinking. That sort of thing.